java/TypeQL.java

```diff
@@ -54,7 +59,7 @@
 import static com.vaticle.typeql.lang.common.TypeQLToken.Predicate.SubString.CONTAINS;
 import static com.vaticle.typeql.lang.common.TypeQLToken.Predicate.SubString.LIKE;
 import static com.vaticle.typeql.lang.common.exception.ErrorMessage.ILLEGAL_CHAR_IN_LABEL;
-import static com.vaticle.typeql.lang.pattern.variable.UnboundVariable.hidden;
+import static com.vaticle.typeql.lang.pattern.variable.UnboundDollarVariable.hidden;
```

Comment:
Most changes in this file are a result of 
- Replacing UnboundVariable with UnboundDollarVariable
- Renaming ThingConstraint.Value to ThingConstraint.Predicate

Others are pointed out with comments